File: backend/src/main/java/pl/kamilwojcik/passwordkeeper/users/api/dto/CreateUserRequest.java
package pl.kamilwojcik.passwordkeeper.users.api.dto;

import pl.kamilwojcik.passwordkeeper.validation.password.app.AppPassword;
import pl.kamilwojcik.passwordkeeper.validation.username.ValidUsername;

import javax.validation.constraints.Email;

public record CreateUserRequest(
        @ValidUsername String username,
        @Email String email,
        @AppPassword String password,
        @AppPassword String storagePassword
) {
}

File: frontend/src/logic/user-api/CreateUserRequest.ts
export class CreateUserRequest {
    public username: string;
    public email: string;
    public password: string;
    public storagePassword: string;


    constructor(username: string, email: string, password: string, storagePassword: string) {
        this.username = username;
        this.email = email;
        this.password = password;
        this.storagePassword = storagePassword;
    }
}